lib/seahorn/BvOpSem2FatMemMgr.cc

```diff
@@ -73,76 +101,92 @@ class FatMemManager : public OpSemMemManager {
   ~FatMemManager() override = default;
 
   FatPtrTy ptrSort() const override {
-    return sort::structTy(m_mem.ptrSort(), m_ctx.alu().intTy(g_slotBitWidth),
+    return sort::structTy(m_main.ptrSort(), m_ctx.alu().intTy(g_slotBitWidth),
                           m_ctx.alu().intTy(g_slotBitWidth));
   }
 
   /// \brief Allocates memory on the stack and returns a pointer to it
   /// \param align is requested alignment. If 0, default alignment is used
   FatPtrTy salloc(unsigned bytes, uint32_t align = 0) override {
-    return mkFatPtr(m_mem.salloc(bytes, align));
+    return mkFatPtr(m_main.salloc(bytes, align), m_slot0.salloc(bytes, align),
```

Comment:
Seems wrong. Probaby should be
```
auto ptrV = m_mem.salloc(bytes, align);
m_slot0.salloc(bytes, align); /* to let slot0manager know about allocation */
m_slot1.salloc(bytes, align);
return mkFatPtr(ptrV);
```

I am assuming that `mkFatPtr()` converts a raw pointer into fat with default values (probably 0) for the slots.

Reply:
Acl

Reply:
done